driver-dse/src/main/java/com/datastax/driver/graph/Edge.java

```diff
@@ -0,0 +1,137 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.graph;
+
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+/**
+ * A default representation of an Edge in DSE Graph.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonDeserialize(using = DefaultEdgeDeserializer.class)
+public class Edge {
```

Comment:
Reusing classes from gremlin-driver definitely sounds interesting.

Their ResultSet class is probably too specific for us, but our ResultSet could be an iterable of [Result](http://tinkerpop.incubator.apache.org/javadocs/3.1.1-SNAPSHOT/full/org/apache/tinkerpop/gremlin/driver/Result.html). There's also a [GraphSONMapper](http://tinkerpop.incubator.apache.org/javadocs/3.1.1-SNAPSHOT/full/org/apache/tinkerpop/gremlin/structure/io/graphson/GraphSONMapper.html) that seems to provide out-of-the-box deserialization of classes like Vertex and Edge.

One hurdle is that gremlin-driver compiles to Java 8, so we'd have to raise the requirement on our dse module.